templates/types/streaming/fastapi/app/engine/generate.py

```diff
@@ -54,14 +57,24 @@ def generate_datasource():
     ingestion_pipeline.vector_store = vector_store
 
     # Run the ingestion pipeline and store the results
-    ingestion_pipeline.run(show_progress=True, documents=documents)
+    nodes = ingestion_pipeline.run(show_progress=True, documents=documents)
 
-    # Default vector store only keeps data in memory, so we need to persist it
-    # Can remove if using a different vector store
+    # SimpleVectorStore keeps the data in memory only - needs to be persisted explicitly
+    # Can be removed if using a different vector store
     if isinstance(vector_store, SimpleVectorStore):
-        vector_store.persist(os.path.join(STORAGE_DIR, "vector_store.json"))
-    # Persist the docstore to apply ingestion strategy
-    docstore.persist(os.path.join(STORAGE_DIR, "docstore.json"))
+        index = VectorStoreIndex(
+            nodes=nodes,
+            storage_context=StorageContext.from_defaults(
+                docstore=docstore,
+                vector_store=vector_store,
+            ),
+            store_nodes_override=True,
+        )
+        # SimpleDocumentStore keeps the data in memory only - needs to be persisted explicitly
```

Comment:
@leehuwuj you made a mistake appying this change from me

Reply:
great catch! i just applied the suggestion automatically to the wrong line, let me update it again